modules/account-lib/src/coin/hbar/transferBuilder.ts

```diff
@@ -0,0 +1,124 @@
+    this.transaction = await super.buildImplementation();
+    return this.transaction;
+  }
+
+  private buildTransferData(): proto.ITransferList {
+    return {
+      accountAmounts: [
+        { accountID: this.buildAccountData(this._source.address), amount: Long.fromString(this._amount).negate() }, // sender
+        { accountID: this.buildAccountData(this._toAddress), amount: Long.fromString(this._amount) }, // recipient
+      ],
+    };
+  }
+
+  private buildAccountData(address: string): proto.AccountID {
+    const accountData = new AccountId(address);
+    return new proto.AccountID({
+      accountNum: accountData.account,
+      realmNum: accountData.realm,
+      shardNum: accountData.shard,
+    });
+  }
+
+  /** @inheritdoc */
+  protected initBuilder(tx: Transaction): void {
+    super.initBuilder(tx);
+    const transferData = tx.txBody.cryptoTransfer;
+    if (transferData && transferData.transfers && transferData.transfers.accountAmounts) {
+      this.initTransfers(transferData.transfers.accountAmounts);
+    }
+  }
+
+  /**
+   * Initialize the transfer specific data, getting the recipient account
+   * represented by the element with a positive amount on the transfer element.
+   * The negative amount represents the source account so it's ignored.
+   *
+   * @param {proto.IAccountAmount[]} transfers array of objects which contains accountID and transferred amount
+   */
+  protected initTransfers(transfers: proto.IAccountAmount[]): void {
+    transfers.forEach(transferData => {
+      const amount = Long.fromValue(transferData.amount!);
+      if (amount.isPositive()) {
+        this.to(stringifyAccountId(transferData.accountID!));
+        this.amount(amount.toString());
+      }
+    });
+  }
+
+  /** @inheritdoc */
+  protected signImplementation(key: BaseKey): Transaction {
+    if (this._multiSignerKeyPairs.length >= DEFAULT_M) {
+      throw new SigningError('A maximum of ' + DEFAULT_M + ' can sign the transaction.');
+    }
+    return super.signImplementation(key);
+  }
+
+  //region Transfer fields
+  /**
+   * Set the destination address where the funds will be sent,
+   * it may take the format `'<shard>.<realm>.<account>'` or `'<account>'`
+   *
+   * @param {string} address the address to transfer funds to
+   * @returns {TransferBuilder} the builder with the new parameter set
+   */
+  to(address: string): this {
+    if (!isValidAddress(address)) {
+      throw new InvalidParameterValueError('Invalid address');
+    }
+    this._toAddress = address;
+    return this;
+  }
+
+  /**
+   * Set the amount to be transferred
+   *
+   * @param {string} amount amount to transfer in tinyBars (there are 100,000,000 tinyBars in one Hbar)
+   * @returns {TransferBuilder} the builder with the new parameter set
+   */
+  amount(amount: string): this {
+    if (!isValidAmount(amount)) {
+      throw new InvalidParameterValueError('Invalid amount');
+    }
+    this._amount = amount;
+    return this;

```

Comment:
I don't think we allow valueless transfers, please add a check here so it's > 0

Reply:
For ethereum Mark confirmed that within account-lib transactions could have an amount of zero. We even added a [ticket ](https://simtlix.atlassian.net/browse/BGA-206) to support that. Could @argjv  and @marktoda  confirm if this is correct? Maybe it was something specific for the smart contract.

Reply:
That is correct for ETH and ETHlike coins, but I do not know about hbar

Reply:
The only things I could think of would be erc20 txes (which I don't know how they work) or AccountCreateTransaction (which could be 0-valued). I think the risk is minimal for now